app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.java

```diff
@@ -158,6 +158,27 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     return llLayout;
   }
 
+  @Override
+  public void onStop() {
+    if (mBound && super.getActivity() != null) {
+      super.getActivity().unbindService(mConnection.downloadServiceInterface);
+      mBound = false;
+    }
+    if (isReceiverRegistered) {
+      faActivity.unregisterReceiver(networkBroadcastReceiver);
+      isReceiverRegistered = false;
+    }
+    super.onStop();
+  }
+
+  @Override
+  public void onDetach() {
+    if (mBound && super.getActivity() != null) {
+      super.getActivity().unbindService(mConnection.downloadServiceInterface);
+      mBound = false;
```

Comment:
```onDetach``` gets called everytime the fragment is recreated inside the Activity. If you're unbinding the service in the ```onDetach``` then you don't need to do it ```onStop``` i.e. it should only be in one method.

Reply:
Oh Yeah. Sorry for that mistake. Lemme correct it.